### PR TITLE
Migrate to Maven Central Publishing 📦 

### PR DESCRIPTION
**Pull request by jumaallan**
### **User description**
Story: https://app.shortcut.com/smileid/story/xxx

## Summary

A few sentences/bullet points about the changes

## Known Issues

Any shortcomings in your work. This may include corner cases not correctly handled or issues related
to but not within the scope of your PR. Design compromises should be discussed here if they were not
already discussed above.

## Test Instructions

Concise test instructions on how to verify that your feature works as intended. This should include
changes to the development environment (if applicable) and all commands needed to run your work.

## Screenshot

If applicable (e.g. UI changes), add screenshots to help explain your work.

___

### **PR Type**
Enhancement

___

### **Description**
- Update Maven Central publishing configuration

- Switch to SonatypeHost.CENTRAL_PORTAL

- Fix version string handling

- Simplify coordinates configuration

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>lib.gradle.kts</strong><dd><code>Update Maven Central publishing configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

lib/lib.gradle.kts

<li>Added import for SonatypeHost from vanniktech maven publish<br> <li> Updated publishToMavenCentral to use SonatypeHost.CENTRAL_PORTAL<br> <li> Fixed version string handling with explicit toString()<br> <li> Simplified coordinates configuration by removing redundant <br>project.version

</details>

  </td>
  <td><a href="https://github.com/smileidentity/android/pull/584/files#diff-9d62a200b35918c059dbb614af706a76e233386ff0bbdd8f7f45795c03f3e737">+7/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about PR-Agent usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

### CODE REVIEW

**Review by robin-smileid — APPROVED** [review]
Changes look fine to me.

File: lib/lib.gradle.kts [hunk 1]
```diff
@@ -89,9 +92,9 @@ moshi {
 }
 
 mavenPublishing {
-    publishToMavenCentral(automaticRelease = true)
+    publishToMavenCentral(SonatypeHost.CENTRAL_PORTAL, automaticRelease = true)
     signAllPublications()
-    coordinates(groupId, artifactId, project.version.toString())
+    coordinates(groupId, artifactId)
```
Comment (robin-smileid) [inline lib/lib.gradle.kts:97]: Do you need to set this to 
```suggestion
    coordinates(groupId, artifactId, project.version)
```

Comment (jumaallan) [reply lib/lib.gradle.kts:97]: Not needed anymore
